akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -469,6 +471,8 @@ final class Broadcast[T](private val outputPorts: Int, eagerCancel: Boolean) ext
 
 object Partition {
 
+  case class PartitionOutOfBoundsException(msg:String) extends RuntimeException(msg) with NoStackTrace
```

Comment:
Just use `expectError()` which return the Throwable, then you can look at the message or call an `isInstanceOf`